src/zmalloc.c

```diff
@@ -123,14 +123,14 @@ void *ztrymalloc_usable(size_t size, size_t *usable) {
 }
 
 /* Allocate memory or panic */
-void *zmalloc(size_t size) {
+__attribute__((flatten)) void *zmalloc(size_t size) {
     void *ptr = ztrymalloc_usable(size, NULL);
     if (!ptr) zmalloc_oom_handler(size);
```

Comment:
that can't obviously cause the zmalloc_oom_handler (being a function pointer) be inlined. as desired (can be huge)

Reply:
no indeed you re right.